### ISSUE TITLE
Feature Request: Show download URL in the description. 

### ISSUE THREAD

**Issue by Abhinav1217** [srevinsaju/zap#51]
Currently the description has following content,
```
Name of Appimage, 
submitted by <User>
<description if provided> else blank
```
Which means information provided in the description is the only source of information on the application. If information is badly worded, misguided or just absent, user have no easy way to know more about the app.

Example of bad source of information:
![Screenshot from 2021-11-09 10-06-08](https://user-images.githubusercontent.com/740584/140863284-8c9521d3-113a-4d2f-9d43-f29643ba4dd8.png)

The first page of google search for spectrum will show information about cable company. Searching spectrum appimage showed the spectron electron js testing framework for me. I agree this is the worst case scenario where developers put fancy taglines instead of clearly telling what application is ( this is a deprecated chat app which is now archived as read-only. ).

It is same problem with the appimagehub website too, but at least on the website, I could click the link to go to the repo and explore what it is.

**Request:**
If it is possible, can we display url for homepage, or where the appimage will be downloaded from, so that we as a user know what will be installed. Currently while downloading, we don't even know what version will be installed unless it is provided in the description by the builder. All other metadata will depend on what is provided by the actual app maintainer, but at the very least, download url can still be displayed by package manager.

**Benifits:**

- Most of the Appimages are community build. Which means some may get outdated while there are other community appimage available which is more up-to-date, or there are multiple people giving appimage of same application. If a download url is provided, user can know which appimage will be installed, they can choose to look up some other community build if they feel like they have a better  or use build provided someone they have used in past and hence trust more. 

- A lot of appimages listed doesn't have enough information in the description ( some doesn't have description at all ). If a url is provided, vigilant user will have something to go on. They can use that url as starting point to find out more about the application such as features or capabilities, and search for alternatives. Even if it is just a download url, it would help to make a more informed decision.

**Comment by srevinsaju** [comment]
Agreed, it would be certainly better to show the URL to the source too, good Feature request!

**Comment by srevinsaju** [comment]
Added a feature in https://github.com/srevinsaju/zap/commit/b8ed33098960119273aea738197c740edb680b60 which shows links to appimage's source code, or their download URLs in zap search command 

Sample screenshot 
![image](https://user-images.githubusercontent.com/48695438/147604980-8c01449d-4a43-490a-ace3-6a0e96f81703.png)

**Comment by Abhinav1217** [comment]
I just tested it, it is immediately noticeable improvement. I now know that zoom was installed from probono repo, and jumping back to spectrum app, It immediately takes me to the github page. 

There is a bug that I encountered testing this feature, I don't know if it is a technical limitation of a TUI  or whether it can be solved. This is why I am not opening a new bug report without discussing here. 

When I `ctrl+click` the link as shown below
![Screenshot from 2021-12-29 16-08-11](https://user-images.githubusercontent.com/740584/147653781-5e31e497-70f8-4bfa-b148-1c716801d78a.png)
i.e when the link  is truncated because of lack of space ( instead of text-wrapping on next line )

 
The browser receives this 

![Screenshot from 2021-12-29 16-09-12](https://user-images.githubusercontent.com/740584/147653873-f844465c-a4a8-4ed5-a125-ccbe8f36f70e.png)

I have tested this in gnome-terminal. alacritty, and kitty. Same result.  When There is enough space to display entire link, it works properly. It may cause issues on some apps which uses hashes or long url path for their downloads.

I don't have any Idea if links can be wrapped to next link in the golang, I know it is possible in npm world and in bash also.
Is there any specific TUI framework that you are using?

**Comment by Abhinav1217** [comment]
Let me know if you want me to open a new issue report regarding this?

**Comment by srevinsaju** [comment]
Yea, you got a point. Can you point me to any repository (regardless of the programming language) where wrapped links work? I was thinking its a limitation of TUIs, but if it works in any other programming language, it should be possible in the library I am using (it's called go-fuzzysearch).

**Comment by Abhinav1217** [comment]
> I was thinking its a limitation of TUIs, but if it works in any other programming language, it should be possible 

At the time of writing that comment, I was mainly thinking about how when you install npm packages, full url are wrapped if space is limited. Also if you just paste a long text or url on your terminal and reduce its width, it will wrap unless it is disabled in settings. Most terminals have text-wrap enabled as default.  

![Peek 2021-12-29 21-39](https://user-images.githubusercontent.com/740584/147682381-d7dc2d4a-7dfe-42f8-b351-73f10527947a.gif)


Simple google search on "TUI text-wrap" provides[ link to a closed issue](https://github.com/jarun/nnn/issues/1225) in `nnn` which is a tui based file manager.  It is mainly written in C. 

I daily use, ranger as file manager, bottom as process viewer, moc for music, and npm along with few other development tools. Text-wrapping works almost universally.  MOC has an option to wrap text or not in settings, as does vim. So we can safely assume it isn't something technically limited on commonly used terminal emulators. 

>  (it's called go-fuzzysearch).

It is a fuzzy search library, not a TUI library.  You can take a look at this go library for creating TUI, called [tview](https://github.com/rivo/tview/search?q=wrap)  . If you weren't using any TUI library before, then you would probably need to refactor a lot, maybe even start from scratch on a separate branch.

**Comment by srevinsaju** [comment]
Thank you, I don't understand how that would help. I can wrap links, and I can wrap text. but, will the wrapped text be clickable? This is something which I am interested to know. 

I am interested to know if something like 
```
some random text                                                | https://github.com/srevinsaju/za
more random text here                                           | p/issues 
```
would work?

**Comment by srevinsaju** [comment]
Yes, but I don't see two virtual terminal panes. In zap, you will be able to see the left virtual pane showing app names, and the right virtual pane showing the description. I am sure the link will be clickable if we didnt have the panes. But, on the brighter side, I think I should rather show the URL on the terminal, once the app is selected, and the enter key is pressed. As of now, it doesn't show the URLs, but only the description. This is something we can work on.

**Comment by Abhinav1217** [comment]
> This is something we can work on.

Sure, until there is some better solution for this. I just completed learning go language, and now I am learning building TUI with [tview](https://github.com/rivo/tview). Maybe in few months I can contribute something here myself.
